### ISSUE TITLE
bwd_kernel_fuse builds its dK/dV-phase dO pointers from Out's strides, so it is only correct when Out and dO share a layout

### ISSUE THREAD

**Issue by jeffdaily** [ROCm/aotriton#236]
In `modules/flash/kernel/bwd_kernel_fuse.py` (main, line 502; 0.13b line 560, same in `tritonsrc/bwd_kernel_fuse.py`) the dK/dV phase constructs the dO pointers with the `Out` strides:

```python
do_ptrs0, do_ptrs1, do_ptrs2 = composed_ptrs(DO,
                                             stride_oz, stride_oh, stride_om, stride_ok,
                                             batch_index, off_h_q, cu_seqlens_q_start + offs_q,
                                             BLOCK_DMODEL0, BLOCK_DMODEL1, BLOCK_DMODEL2)
```

The dQ phase (line 184 / 213) uses `stride_do*` for the same tensor. The `attn_bwd_params` API takes separate `TensorView`s with their own strides for `Out` and `DO`, so callers can legitimately pass them in different layouts; when they do, the dK/dV phase reads dO through the wrong strides (in bounds: wrong dK/dV; out of bounds: memory fault).

Reproducer (PyTorch nightly on ROCm 10.0 / gfx950, AOTriton 0.13b, before pytorch's workaround): `_scaled_dot_product_efficient_attention_backward` passes dO as a (batch, seq, heads, dim)-contiguous copy; the forward returns `out` in the same layout, so the common path is fine. Handing the backward `out.contiguous()` (a (batch, heads, seq, dim)-contiguous tensor, which is what context-parallel ring attention produces) memory-faults in `bwd_kernel_fuse` at 8x8x4096x32 fp32 causal, with a contiguous or a zero-stride grad_out alike; re-laying `out` out as (batch, seq, heads, dim)-contiguous makes the same call pass.

```python
import torch
B, H, S, D = 8, 8, 4096, 32
q = torch.randn(B, H, S, D, device="cuda"); k = torch.randn_like(q); v = torch.randn_like(q)
out, lse, seed, offset = torch.ops.aten._scaled_dot_product_efficient_attention(q, k, v, None, True, 0.0, True)
go = torch.ones(B, H, S, D, device="cuda")
o = out.contiguous()   # (B, H, S, D)-contiguous; out itself is (B, S, H, D)-contiguous and passes
torch.ops.aten._scaled_dot_product_efficient_attention_backward(go, q, k, v, None, o, lse, seed, offset, 0.0, [True, True, True, False], True)
torch.cuda.synchronize()
```

Expected: the kernel honors `stride_do*` for `DO` in both phases (or the API documents that `Out` and `DO` must share a layout and the shim checks it). PyTorch is working around it by materializing `out` in dO's layout before the call (pytorch PR to follow).

This issue was drafted with the assistance of an AI agent (Claude); the code reading and the reproducer were verified by hand.
